doc/ansible/dev/developer_guide.md

```diff
@@ -0,0 +1,430 @@
+         
+INFO[0000] Watching foo.example.com/v1alpha1, Foo, default 
+```
+
+Now that the operator is watching resource `Foo` for events, the creation of a
+Custom Resource will trigger our Ansible Role to be executed. Take a look at
+`deploy/cr.yaml`:
+```yaml
+apiVersion: "foo.example.com/v1alpha1"
+kind: "Foo"
+metadata:
+  name: "example"
+```
+
+Since `spec` is not set, Ansible is invoked with no extra variables. The next
+section covers how extra variables are passed from a Custom Resource to
+Ansible. This is why it is important to set sane defaults for the operator.
+
+Create a Custom Resource instance of Foo with default var `state` set to
+`present`:
+```bash
+$ kubectl create -f deploy/cr.yaml
+```
+
+Check that namespace `test` was created:
+```bash
+$ kubectl get namespace
+NAME          STATUS    AGE
+default       Active    28d
+kube-public   Active    28d
+kube-system   Active    28d
+test          Active    3s
+```
+
+Modify `deploy/cr.yaml` to set `state` to `absent`:
+```yaml
+apiVersion: "foo.example.com/v1alpha1"
+kind: "Foo"
+metadata:
+  name: "example"
+spec:
+  state: "absent"
+```
+
+Apply the changes to Kubernetes and confirm that the namespace is deleted:
+```bash
+$ kubectl apply -f deploy/cr.yaml
+$ kubectl get namespace
+NAME          STATUS    AGE
+default       Active    28d
+kube-public   Active    28d
+kube-system   Active    28d
+```
+
+### Testing an Ansible operator on a cluster
+
+Now that a developer is confident in the operator logic, testing the operator
+inside of a pod on a Kubernetes cluster is desired. Running as a pod inside a
+Kubernetes cluster is preferred for production use.
+
+To build the `foo-operator` image and push it to a registry:
+```
+$ operator-sdk build quay.io/example/foo-operator:v0.0.1
+$ docker push quay.io/example/foo-operator:v0.0.1
+```
+
+Kubernetes deployment manifests are generated in `deploy/operator.yaml`. The
+deployment image in this file needs to be modified from the placeholder
+`REPLACE_IMAGE` to the previous built image. To do this run:
+```
+$ sed -i 's|REPLACE_IMAGE|quay.io/example/foo-operator:v0.0.1|g' deploy/operator.yaml
+```
+
+Deploy the foo-operator:
+
+```sh
+$ kubectl create -f deploy/rbac.yaml
+$ kubectl create -f deploy/operator.yaml
+```
+
+**NOTE**: `deploy/rbac.yaml` creates a `ClusterRoleBinding` and assumes we are
+working in namespace `default`. If you are working in a different namespace you
+must modify this file before creating it.
+
+Verify that the foo-operator is up and running:
+
+```sh
+$ kubectl get deployment
+NAME                     DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+foo-operator       1         1         1            1           1m
+```
+
+## Extra vars sent to Ansible
+The extravars that are sent to Ansible are predefined and managed by the operator. The `spec` section will pass along the key-value pairs as extra vars. This is equivalent to how above extra vars are passed in to `ansible-playbook`.
```

Comment:
What do you mean that they are "predefined and managed by the operator"? Is that something different than them being set in the CR's spec? If I'm a new dev reading this, I would take "predefined" to mean I can't change it, but I think this section is intended to be describing how to set these values.

If you are trying to describe values that are pre-defined, that the end user cannot control through the CR, I suggest being more explicit about describing them an contrasting that with the ability to set values in the CR.

Reply:
If by "predefined" you just mean that the user can define them in the CR, I suggest not using the word "predefined".